python/ray/train/tests/test_gpu.py

```diff
@@ -100,6 +100,52 @@ def train_fn():
         )
 
 
+@pytest.mark.parametrize("cuda_visible_devices", ["", "1,2"])
+@pytest.mark.parametrize("num_gpus_per_worker", [0.5, 1, 2])
+def test_torch_get_amd_device(
+    shutdown_only, num_gpus_per_worker, cuda_visible_devices, monkeypatch, tmp_path
+):
+    if cuda_visible_devices:
+        # Test if `get_device` is correct even with user specified env var.
+        monkeypatch.setenv("ROCR_VISIBLE_DEVICES", cuda_visible_devices)
+
+    ray.init(num_cpus=4)
```

Comment:
I am not sure if it runnable if we start a ray cluster with 4 CPUs but request GPUs for `TorchTrainer`.